spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/client/ServiceCombClient.java

```diff
@@ -411,12 +411,13 @@ public HeardBeatStatus heartbeat(HeartbeatRequest heartbeatRequest) throws Servi
       if (response.getStatusCode() == HttpStatus.SC_OK) {
         LOGGER.debug("heartbeat success.");
         return HeardBeatStatus.SUCCESS;
-      } else {
+      } else if (response.getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
```

Comment:
I found heartbeat is call in an scheduler. And don't understand why this can 'avoid too many heartbeat.

```  
 ScheduledFuture currentTask = this.scheduler
        .scheduleWithFixedDelay(() -> {
```

Reply:
in some case it trigger flow limiting:
if heard beat failed , will retry -> retry success, heard beat -> heard beat failed , will retry .
it's a loop.
so I add a condition : retry only if the service does not exist